services/workflows-service/src/workflow/workflow.service.ts

```diff
@@ -648,38 +611,124 @@ export class WorkflowService {
 
       throw new BadRequestException(`Invalid decision status: ${status}`);
     })();
-    const documentsWithDecision = runtimeData?.context?.documents?.map(
-      (document: DefaultContextSchema['documents'][number]) => {
-        if (document.id !== documentId) return document;
+    const document = runtimeData?.context?.documents?.find(
+      (document: DefaultContextSchema['documents'][number]) => document.id === documentId,
+    );
 
-        return {
-          id: document.id,
-          decision: {
-            ...newDecision,
-            status,
-          },
-        };
+    const documentWithDecision = {
+      ...document,
+      id: document.id,
+      decision: {
+        ...newDecision,
+        status,
       },
+    };
+    const checkRequiredFields = status === 'approved' ? true : false;
+    const updatedWorkflow = await this.updateDocumentById(
+      { workflowId, documentId, checkRequiredFields },
+      documentWithDecision as unknown as DefaultContextSchema['documents'][number],
+    );
+
+    return updatedWorkflow;
+  }
+  async updateDocumentById(
+    {
+      workflowId,
+      documentId,
+      checkRequiredFields = true,
+    }: {
+      workflowId: string;
+      documentId: string;
+      checkRequiredFields?: boolean;
+    },
+    data: DefaultContextSchema['documents'][number] & { propertiesSchema?: object },
```

Comment:
90% of the time using `Record` over `object` is the way to go. `object` disallows using properties on the object (unless runtime checks are made).